File: 프로그래머스/2/12953. N개의 최소공배수/N개의 최소공배수.cpp
#include <string>
#include <vector>

using namespace std;

int euclid(int a, int b)
{
    if (a % b == 0) return b;

    return euclid(b, a % b);
}
int lcm(int a, int b)
{
    return a * b / euclid(a, b);
}
int solution(vector<int> arr)
{
    int a = lcm(arr[0], arr[1]);
    
    for (int i = 2; i < arr.size(); i++)
        a = lcm(a, arr[i]);
    
    return a;
}

File: 프로그래머스/2/12953. N개의 최소공배수/N개의 최소공배수.java
class Solution {
    
    private int lcm(int a, int b) {
        return a * b / this.gcd(a, b);
    }
    
    private int gcd(int a, int b) {
        if (b == 0) return a;
        
        return this.gcd(b, a % b);
    }
    
    public int solution(int[] arr) {
        int answer = 1;
        
        for (int i : arr) {
            answer = this.lcm(answer, i);
        }
        
        return answer;
    }
}